src/utils/filePathGenerator.ts

```diff
@@ -29,12 +29,27 @@ const registryAccess = new RegistryAccess();
  * @param packageDir optional package directory to apply to the file paths
  * @returns array of file paths
  */
+// eslint-disable-next-line complexity
 export const filePathsFromMetadataComponent = (
   { fullName, type }: MetadataComponent,
   packageDir?: string
 ): string[] => {
   const packageDirWithTypeDir = packageDir ? join(packageDir, type.directoryName) : type.directoryName;
 
+  if (type.strategies?.adapter === 'digitalExperience') {
+    // child MD Type, the metafile is a JSON, not an XML
+    if (['digitalexperience'].includes(type.id)) {
+      return [
+        join(packageDirWithTypeDir, `${fullName.split('.')[0]}${sep}${fullName.split('.')[1]}${sep}${META_JSON_FILE}`),
+      ];
+    }
+
+    // parent MD Type
+    if (['digitalexperiencebundle'].includes(type.id)) {
```

Comment:
why not just if (type.id === 'digitalexperiencebundle') {

Reply:
I used .includes to indicate that more types like DEB/DE (from the same source adapter) can be added to the array in the future and I also saw a similar check in the original PR. I didn't want to do it this way either, I'll change it.